domain/src/Aggregates/SharePoolingAsset/ValueObjects/SharePoolingAssetId.php

```diff
@@ -4,16 +4,21 @@
 
 namespace Domain\Aggregates\SharePoolingAsset\ValueObjects;
 
+use Domain\Aggregates\SharePoolingAsset\ValueObjects\Exceptions\SharePoolingAssetIdException;
 use Domain\ValueObjects\AggregateRootId;
 use Domain\ValueObjects\Asset;
-use Ramsey\Uuid\Uuid;
 
-final class SharePoolingAssetId extends AggregateRootId
+final readonly class SharePoolingAssetId extends AggregateRootId
 {
-    private const NAMESPACE = '94ff3977-46f5-4efe-8a89-e474d375232f';
-
     public static function fromAsset(Asset $asset): static
     {
-        return self::fromString(Uuid::uuid5(self::NAMESPACE, (string) $asset)->toString());
+        ! $asset->isNonFungible || throw SharePoolingAssetIdException::assetIsNonFungible($asset);
```

Comment:
ℹ️ Same invariant for generating share pooling asset IDs, preventing doing so from a non-fungible asset